godot-core/src/builtin/collections/packed_array.rs

```diff
@@ -433,10 +445,39 @@ macro_rules! impl_packed_array {
             }
         }
 
+        #[doc = concat!("Creates a `", stringify!($PackedArray), "` from the given Rust array.")]
+        impl<const N: usize> From<[$Element; N]> for $PackedArray {
+            fn from(arr: [$Element; N]) -> Self {
+                let mut ret = Self::new();
+                if N == 0 {
+                    return ret;
+                }
+                ret.resize(N);
+                // SAFETY: Data is moved and the original array is forgotten.
+                unsafe {
+                    let arr = std::mem::ManuallyDrop::new(arr);
+                    ret.move_from_slice(arr.as_ptr(), N);
+                }
+                ret
+            }
+        }
```

Comment:
There is a lot of duplicate code for slices, arrays and `Vec`. Could you consolidate it somehow?

Reply:
That kinda requires generic functions. I'll see what i can do.

Reply:
No need for generic, i added new constructor and simplify alot of the construction.